source/codegen/templates/service_helpers.mako

```diff
@@ -129,36 +126,46 @@ ${initialize_standard_input_param(function_name, parameter)}\
 ## Initialize an enum input parameter for an API call.
 <%def name="initialize_enum_input_param(function_name, parameter)">\
 <%
+  config = data['config']
+  enums = data['enums']
+  namespace_prefix = config["namespace_component"] + "_grpc::"
   parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
-  pascal_parameter_name = common_helpers.snake_to_pascal(parameter_name)
+  field_name = common_helpers.camel_to_snake(parameter["name"])
+  pascal_field_name = common_helpers.snake_to_pascal(field_name)
   map_name = parameter["enum"].lower() + "_input_map_"
-  iterator_name = parameter_name + "_imap_it"
-  enum_type_prefix = function_name + "Request::" + pascal_parameter_name + "EnumCase::"
-  param_all_caps_snake = parameter_name.upper()
+  iterator_name = field_name + "_imap_it"
+  one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{pascal_field_name}EnumCase'
+  enum_request_snippet = f'request->{field_name}()'
+  raw_request_snippet = f'request->{field_name}_raw()'
 %>\
       ${parameter['type']} ${parameter_name};
-      switch (request->${parameter_name}_enum_case()) {
-        case ${enum_type_prefix}k${pascal_parameter_name}: {
-          auto ${iterator_name} = ${map_name}.find(request->${parameter_name}());
+      switch (request->${field_name}_enum_case()) {
+        case ${one_of_case_prefix}::k${pascal_field_name}: {
+% if enums[parameter["enum"]].get("generate-mappings", False):
+          auto ${iterator_name} = ${map_name}.find(${enum_request_snippet});
           if (${iterator_name} == ${map_name}.end()) {
             return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for ${parameter_name} was not specified or out of range.");
           }
+<%
+enum_request_snippet = f'{iterator_name}->second'
```

Comment:
Consider:
`enum_request_snippet`  is being reassigned in a style that's usually OK:
```c++
auto str = "one thing";
if (generate_mappings)
  str = "some other thing";
```

But it's a lot harder to read in mako. 

Can you give `iterator_name` a better name? `mapped_enum_request_iterator_name`?

Or Consider:
Would it work to pull it up into the python block?

```python
generate_mappings = enums[parameter["enum"]].get("generate-mappings", False)
direct_enum_request_snippet = f'request->{field_name}()'
mapped_enum_request_snippet =  f'{iterator_name}->second'
enum_request_snippet = mapped_enum_request_snippet if generate_mappings else direct_enum_request_snippet 
````

Reply:
> Consider:
enum_request_snippet is being reassigned in a style that's usually OK:

I have a follow up request where we will have 3 variants of this `enum_request_snippet`, `mapped_enum_request_snippet` and `raw_request_snippet`, where this reassignment is not required. I am leaving it as is for now.

> Can you give iterator_name a better name? mapped_enum_request_iterator_name?

Updated to `mapped_enum_iterator_name` in both `initialize_enum_input_param` and `set_response_values` functions.

>  Would it work to pull it up into the python block?

I couldn't update `enum_request_snippet` directly based on `generate_mappings`, because I still need original snippet to use in `map_name.find(enum_request_snippet)` before reassigning the variable to use `mapped_enum_request_snippet`. Irrespective, this code is further refactored in subsequent PR to consider 3 cases, I can try to figure out best way to pull out pieces of code into python block.